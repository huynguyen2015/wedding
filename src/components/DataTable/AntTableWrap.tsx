import { Table } from 'antd'
import { ColumnsType } from 'antd/lib/table'

export interface IAntTableWrapProps {
  dataSource?: any[]
  columns?: ColumnsType<any>
  isLoading?: boolean
}

const AntTableWrap = ({
  dataSource,
  columns,
  isLoading
}: IAntTableWrapProps) => {
  return (
    <Table
      size="middle"
      className="custom-ant-table"
      rowKey={(record) => record.id || ''}
      columns={columns}
      pagination={false}
      loading={isLoading}
      dataSource={dataSource}
      scroll={{ x: 'max-content', scrollToFirstRowOnChange: true }}
    />
  )
}

export default AntTableWrap
