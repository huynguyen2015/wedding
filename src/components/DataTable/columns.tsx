import { L } from '../../lib/abpUtility'
import { renderDateTime } from '../../lib/helper'
import moment from 'moment-timezone/moment-timezone'

const SystemColumn = {
  title: L('SYSTEM'),
  dataIndex: 'creationTime',
  key: 'creationTime',
  width: 300,
  render: (creationTime, row) => {
    const createdAtAgo = moment(creationTime).fromNow()
    const updatedAtAgo = moment(row.lastModificationTime).fromNow()
    return (
      <div className="text-muted small">
        <div>
          {L('CREATED_BY')} {row.creatorUser?.displayName} {createdAtAgo}
          <br />
          {L('AT')} {renderDateTime(creationTime)}
        </div>
        {row.lastModifierUser && (
          <div>
            {L('UPDATED_BY')} {row.lastModifierUser?.displayName} {updatedAtAgo}
            <br />
            {L('AT')} {renderDateTime(row.lastModificationTime)}
          </div>
        )}
      </div>
    )
  }
}

export default SystemColumn
