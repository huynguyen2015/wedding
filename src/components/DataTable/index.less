.table-header {
  align-items: center;
  .title {
    display: flex;
    align-items: center;
    font-weight: bold;
    font-size: 16px;
    line-height: 24px;
    > h3 {
      margin-bottom: 0;
      text-transform: capitalize;
    }
  }
}
