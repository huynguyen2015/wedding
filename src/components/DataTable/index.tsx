import React from 'react'
import { Button, Card, Col, Pagination, Row } from 'antd'
import { PlusOutlined, ReloadOutlined } from '@ant-design/icons'
import { isGranted, L } from '../../lib/abpUtility'
import './index.less'

export interface IDataTableProps {
  title?: string
  keywordPlaceholder?: string
  textAddNew?: string
  onCreate?: () => void
  onRefresh?: () => void
  pagination?: any
  createPermission?: string
  handleRefresh?: (key, value) => void
  actionComponent?: any
  filterComponent?: any
  children?: any
}

const DataTable: React.FunctionComponent<IDataTableProps> = ({
  title,
  textAddNew,
  onCreate,
  onRefresh,
  pagination,
  createPermission,
  actionComponent,
  filterComponent,
  children
}) => {
  const handleCreate = () => {
    onCreate && onCreate()
  }

  const handleOnChange = (page, pageSize) => {
    if (pagination && pagination.onChange) {
      pagination.onChange({ current: page, pageSize: pageSize })
    }
  }

  return (
    <>
      <Row className={'mb-2 table-header'} gutter={[8, 8]}>
        {title && <Col className="title">{L(title)}</Col>}
        <Col className="text-right justify-content-end d-flex" flex="auto">
          {actionComponent}
          {onRefresh && (
            <Button
              id="btn-refresh-table"
              type="ghost"
              shape="circle"
              size="large"
              icon={<ReloadOutlined />}
              onClick={onRefresh}
              style={{
                boxShadow: '0px 4px 8px rgba(110, 186, 196, 0.2)',
                marginRight: onCreate ? '10px' : '0'
              }}>
              {textAddNew}
            </Button>
          )}
          {onCreate && (!createPermission || isGranted(createPermission)) && (
            <Button
              id="btn-add-new"
              type="primary"
              shape="circle"
              size="large"
              icon={<PlusOutlined />}
              onClick={handleCreate}
              style={{ boxShadow: '0px 4px 8px rgba(110, 186, 196, 0.2)' }}>
              {textAddNew}
            </Button>
          )}
        </Col>
      </Row>
      <Card bordered={false}>
        <Row gutter={[16, 16]}>
          {filterComponent && (
            <Col sm={{ span: 24, offset: 0 }}>{filterComponent}</Col>
          )}
          <Col sm={{ span: 24, offset: 0 }}>{children}</Col>
        </Row>
        {pagination && pagination.total > 0 && (
          <Row className="mt-3 pb-3">
            <Col sm={{ span: 24, offset: 0 }} style={{ textAlign: 'end' }}>
              <Pagination
                size="small"
                showTotal={(total) => L('TOTAL_{0}_ITEMS', total)}
                pageSizeOptions={[10, 25, 50, 100, 500, 1000]}
                {...pagination}
                onChange={handleOnChange}
              />
            </Col>
          </Row>
        )}
      </Card>
    </>
  )
}

export default DataTable
