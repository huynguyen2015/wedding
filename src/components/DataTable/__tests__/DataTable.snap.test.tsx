import DataTable from '../index'
import renderer from 'react-test-renderer'

describe('DataTable', function () {
  let dataTable
  beforeEach(function () {})

  afterEach(function () {
    dataTable = null
  })

  it('To match snapshot data table', async () => {
    dataTable = renderer
      .create(
        <DataTable
          title={'Data table'}
          textAddNew={'Add new'}
          onCreate={() => {}}
          onRefresh={() => {}}
          pagination={{}}
          filterComponent={'Filter component'}>
          <table></table>
        </DataTable>
      )
      .toJSON()

    expect(dataTable).toMatchSnapshot()
  })
})
