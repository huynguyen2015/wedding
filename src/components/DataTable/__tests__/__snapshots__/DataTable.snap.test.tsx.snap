// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable To match snapshot data table 1`] = `
Array [
  <div
    className="ant-row mb-2 table-header"
    style={
      Object {
        "marginBottom": -4,
        "marginLeft": -4,
        "marginRight": -4,
        "marginTop": -4,
      }
    }
  >
    <div
      className="ant-col title"
      style={
        Object {
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        }
      }
    >
      Data table
    </div>
    <div
      className="ant-col text-right justify-content-end d-flex"
      style={
        Object {
          "flex": "auto",
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        }
      }
    >
      <button
        className="ant-btn ant-btn-circle ant-btn-ghost ant-btn-lg"
        onClick={[Function]}
        style={
          Object {
            "boxShadow": "0px 4px 8px rgba(110, 186, 196, 0.2)",
            "marginRight": "10px",
          }
        }
        type="button"
      >
        <span
          aria-label="reload"
          className="anticon anticon-reload"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="reload"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M909.1 209.3l-56.4 44.1C775.8 155.1 656.2 92 521.9 92 290 92 102.3 279.5 102 511.5 101.7 743.7 289.8 932 521.9 932c181.3 0 335.8-115 394.6-276.1 1.5-4.2-.7-8.9-4.9-10.3l-56.7-19.5a8 8 0 00-10.1 4.8c-1.8 5-3.8 10-5.9 14.9-17.3 41-42.1 77.8-73.7 109.4A344.77 344.77 0 01655.9 829c-42.3 17.9-87.4 27-133.8 27-46.5 0-91.5-9.1-133.8-27A341.5 341.5 0 01279 755.2a342.16 342.16 0 01-73.7-109.4c-17.9-42.4-27-87.4-27-133.9s9.1-91.5 27-133.9c17.3-41 42.1-77.8 73.7-109.4 31.6-31.6 68.4-56.4 109.3-73.8 42.3-17.9 87.4-27 133.8-27 46.5 0 91.5 9.1 133.8 27a341.5 341.5 0 01109.3 73.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.6 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c-.1-6.6-7.8-10.3-13-6.2z"
            />
          </svg>
        </span>
        <span>
          Add new
        </span>
      </button>
      <button
        className="ant-btn ant-btn-circle ant-btn-primary ant-btn-lg"
        onClick={[Function]}
        style={
          Object {
            "boxShadow": "0px 4px 8px rgba(110, 186, 196, 0.2)",
          }
        }
        type="button"
      >
        <span
          aria-label="plus"
          className="anticon anticon-plus"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="plus"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <defs>
              <style />
            </defs>
            <path
              d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
            />
            <path
              d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
            />
          </svg>
        </span>
        <span>
          Add new
        </span>
      </button>
    </div>
  </div>,
  <div
    className="ant-card"
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="ant-row"
        style={
          Object {
            "marginBottom": -8,
            "marginLeft": -8,
            "marginRight": -8,
            "marginTop": -8,
          }
        }
      >
        <div
          className="ant-col ant-col-sm-24 ant-col-sm-offset-0"
          style={
            Object {
              "paddingBottom": 8,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 8,
            }
          }
        >
          Filter component
        </div>
        <div
          className="ant-col ant-col-sm-24 ant-col-sm-offset-0"
          style={
            Object {
              "paddingBottom": 8,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 8,
            }
          }
        >
          <table />
        </div>
      </div>
    </div>
  </div>,
]
`;
